### PR TITLE
Deals Story - Create a new model for Category

### PR DESCRIPTION
**Pull request by DuaaH**

### CODE REVIEW

File: src/routes/routes.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import authRouter from '@/modules/auth/auth.routes';
+import categoryRouter from '@/modules/category/category.routes';
 import userRouter from '@/modules/user/user.routes';
 import express from 'express';
 
 const router = express.Router();
 console.log('inside router');
 router.use('/auth', authRouter);
 router.use('/user', userRouter);
+router.use('/', categoryRouter);
```
Comment (muradrajab) [inline src/routes/routes.ts:10]: router.use('/category', categoryRouter);

File: src/modules/category/category.routes.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import { addCategoryController, getCategoriesController } from './category.controller';
+import { authMiddleware } from '@/middlewares/auth.middleware';
+
+const categoryRouter = express.Router();
+
+// Route to fetch all categories
+categoryRouter.get('/categories', authMiddleware, getCategoriesController);
+
+// Route to add a new category
+categoryRouter.post('/categories', authMiddleware, addCategoryController);
```
Comment (muradrajab) [inline src/modules/category/category.routes.ts:11]: categoryRouter.post('/', authMiddleware, addCategoryController);

File: src/modules/category/category.repo.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        try {
+            // Fetch all categories from the Category table
+            const categories = await DB.Categories.findAll();
+            return categories;
+        } catch (error) {
+            console.error('Error fetching categories:', error);
+            return null;
+        }
+    },
+     findCategoryByName : async (categoryName: string): Promise<CategoryModel | null> => {
```
Comment (muradrajab) [inline src/modules/category/category.repo.ts:16]: findByName

File: src/modules/category/category.repo.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { DB } from '@/database';
+import { CategoryModel } from '@/database/models/category.model';
+
+
+const repo = {
+    allCategories: async (): Promise<CategoryModel[] | null> => {
```
Comment (muradrajab) [inline src/modules/category/category.repo.ts:6]: getAll, findAll, fetchAll

File: src/modules/category/category.controller.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                message: 'Successfully get categories',
+                data: response
+            });
+        } catch (error) {
+            next(error);
+        }
+    };
+
+// Controller to add a new category
+export const addCategoryController = async (
```
Comment (muradrajab) [inline src/modules/category/category.controller.ts:23]: addCategory

File: src/modules/category/category.controller.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import { NextFunction, Request, Response } from 'express';
+import { addCategoryService, categoryService } from './category.service';
+import { CustomError } from '@/utils/custom-error';
+
+export const getCategoriesController = async (
```
Comment (muradrajab) [inline src/modules/category/category.controller.ts:5]: getCategories

File: src/modules/category/category.service.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import repo from './category.repo';
+import { CustomError } from '@/utils/custom-error';
+
+export const categoryService = async () => {
```
Comment (muradrajab) [inline src/modules/category/category.service.ts:4]: getCategories

File: src/modules/category/category.service.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        console.log('Categories:', categories);
+        return categories;
+    } catch (error) {
+        console.error('Error in category:', error);
+        throw new CustomError('Failed to fetch categories', 500);
+    }
+};
+
+// Add a new category
+export const addCategoryService = async (categoryData: { category_name: string }) => {
```
Comment (muradrajab) [inline src/modules/category/category.service.ts:21]: addCategory

File: src/database/models/category.model.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        autoIncrement: true,
+      },
+      category_name: {
+        type: DataTypes.STRING(255),
+        allowNull: false,
+        unique: true,
+        comment: 'Name of the category',
+      }
```
Comment (muradrajab) [inline src/database/models/category.model.ts:32]: why extra }?

Comment (DuaaH) [reply src/database/models/category.model.ts:32]: what do you mean by extra }
image

**Pull request merged**